### ISSUE TITLE
[Bug] azure api 环境变量不能开箱即用，参数已经设置

### ISSUE THREAD

**Issue by luckfu** [lobehub/lobe-chat#2054]
### 💻 系统环境

Other

### 📦 部署环境

Vercel / Zeabur / Sealos

### 🌐 浏览器

Chrome

### 🐛 问题描述

azure api 环境变量不能开箱即用，参数已经设置
<img width="572" alt="截屏2024-04-15 21 04 23" src="https://github.com/lobehub/lobe-chat/assets/1396910/ced8915c-5afa-4ac6-b192-b2f69537c840">

### 📷 复现步骤

<img width="838" alt="截屏2024-04-15 21 05 15" src="https://github.com/lobehub/lobe-chat/assets/1396910/758cb43d-00c5-491a-9542-063fabe142c8">

**Comment by arvinxx** [comment]
你选azure 的模型了吗

**Comment by henryoln** [comment]
环境变量配置错了，应该是AZURE_MODEL_LIST和AZURE_API_KEY

**Comment by Elity** [comment]
I have configured AZURE_ENDPOINT, AZURE_MODEL_LIST, AZURE_API_VERSION, and AZURE_API_KEY, but I still get the some error:
![image](https://github.com/lobehub/lobe-chat/assets/3899648/6fc74933-576a-47d0-82f7-fb852c5bbf7f)

**Comment by Elity** [comment]
> I have configured AZURE_ENDPOINT, AZURE_MODEL_LIST, AZURE_API_VERSION, and AZURE_API_KEY, but I still get the some error: ![image](https://private-user-images.githubusercontent.com/3899648/322691747-6fc74933-576a-47d0-82f7-fb852c5bbf7f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMyMzg0NzUsIm5iZiI6MTcxMzIzODE3NSwicGF0aCI6Ii8zODk5NjQ4LzMyMjY5MTc0Ny02ZmM3NDkzMy01NzZhLTQ3ZDAtODJmNy1mYjg1MmM1YmJmN2YucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MTZUMDMyOTM1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWI1NThjNGQ2ZTkyM2Y1OGNlZWQ1MzQwYmM1ZGE3ODlhZGU3N2E1NjBmYzE0YmJmMGIwZTEzYzVhZGI0YjhhOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.RtALKefU8CtzLXrsSTYt5L9DrG3S1rTbEVANSuaz794)

re-select the model at frontend page solved the problem:
![image](https://github.com/lobehub/lobe-chat/assets/3899648/09fe2dfb-11e4-4796-8427-83297311a814)

**Comment by amitshuklabag** [comment]
I'm also facing the same issue, After passing the Azure API Key it still asks for an open AI key.

**Comment by cnkang** [comment]
我用了个变通的方法，可以不设置OPENAI的KEY，只使用azure：
USE_AZURE_OPENAI=1
OPENAI_MODEL_LIST=-all

**Comment by kmahyyg** [comment]
Same problem here, after setting Azure OAI following docs, then:

Workaround by adding additional environment variables:
```
USE_AZURE_OPENAI=1
OPENAI_MODEL_LIST=-all
```

**Comment by kmahyyg** [comment]
@arvinxx After I tested at latest release version today, I found the root cause. 
1. Document is NOT updated in latest version, for Azure OAI, deployment name is still required to be the same as model ID in lobechat, no document mentioned that.
2. Model name parsing is not functional, `-all,+gpt-3.5.turbo-1106->gpt-35-turbo-example` will request azure for deployment `gpt-3.5-turbo-1106`, instead of document recorded actual deployment `gpt-35-turbo-example`, thus generating error "deployment not found".
3. Azure OAI available model list is not updated.

Either fix point 2 or fix point 1+3 will solve the problem.

It's still a bug.
